authorization-common/src/main/java/com/sequenceiq/authorization/service/AbstractPermissionCheckerService.java

```diff
@@ -60,34 +65,39 @@ protected Object checkPermission(ProceedingJoinPoint proceedingJoinPoint, Method
         Optional<Annotation> classAnnotation = commonPermissionCheckingUtils.getClassAnnotation(authorizationClass);
 
         AuthorizationResource classAuthorizationResource = (AuthorizationResource) classAnnotation.get();
+        AuthorizationResourceType resource = classAuthorizationResource.type();
+        String userCrn = ThreadBasedUserCrnProvider.getUserCrn();
 
         List<? extends Annotation> annotations = getPossibleMethodAnnotations().stream()
                 .map(c -> methodSignature.getMethod().getAnnotation(c))
                 .collect(Collectors.toList());
 
-        Annotation methodAnnotation = validateNumberOfAnnotations(methodSignature, annotations);
-        if (methodAnnotation instanceof DisableCheckPermissions) {
+        Optional<Annotation> methodAnnotation = validateNumberOfAnnotations(methodSignature, annotations);
+        if (!methodAnnotation.isPresent()) {
+            LOGGER.info("Your Controller ({}) method {} does not have any authorization related annotation, " +
+                            "thus we are checking write permission on current account.",
+                    proceedingJoinPoint.getTarget().getClass().getSimpleName(), methodSignature.getMethod().getName());
+            commonPermissionCheckingUtils.checkPermissionForUser(resource, AuthorizationResourceAction.WRITE, userCrn);
+            return commonPermissionCheckingUtils.proceed(proceedingJoinPoint, methodSignature, startTime);
+        } else if (methodAnnotation.get() instanceof DisableCheckPermissions) {
             return commonPermissionCheckingUtils.proceed(proceedingJoinPoint, methodSignature, startTime);
         }
 
-        String userCrn = ThreadBasedUserCrnProvider.getUserCrn();
-        PermissionChecker<? extends Annotation> permissionChecker = permissionCheckerMap.get(methodAnnotation.annotationType());
-        AuthorizationResourceType resource = classAuthorizationResource.type();
-        return permissionChecker.checkPermissions(methodAnnotation, resource, userCrn, proceedingJoinPoint, methodSignature, startTime);
+        PermissionChecker<? extends Annotation> permissionChecker = permissionCheckerMap.get(methodAnnotation.get().annotationType());
+        return permissionChecker.checkPermissions(methodAnnotation.get(), resource, userCrn, proceedingJoinPoint, methodSignature, startTime);
     }
 
-    private Annotation validateNumberOfAnnotations(MethodSignature methodSignature, List<? extends Annotation> annotations) {
+    private Optional<Annotation> validateNumberOfAnnotations(MethodSignature methodSignature, List<? extends Annotation> annotations) {
```

Comment:
What happens if someone specifies another annotation, should we verify an exact list of Annotations we are looking for? Would it be an error in the permission checking util or cause some RunTimeError?

Reply:
Child class (PermissionCheckerService) has a static list about possible authz annotations and we are checking those.